### PR TITLE
Function tooltip shows documentation for its parameter

### PR DESCRIPTION
**Pull request by mcon**
# Summary
I believe this change should fix this issue https://github.com/dotnet/fsharp/issues/7455, and have confirmed that it appears to do so when running an instance of ReSharper with this change built into it.
The method in question that's been changed `LinesBefore`, was behaving more like `LinesAfter`.

# Testing
 - Manual testing with Rider
 - Added a new testcase which should cover this case, though I've not been able to run it due to a "did not resolve mscorelib" error when trying to run the FCS tests (I'm running on Windows, nothing out of the ordinary) - if someone could point me to how to go about running them, that would be great.

### CODE REVIEW

File: tests/service/MultiProjectAnalysisTests.fs [hunk 1]
```diff
@@ -43,6 +44,12 @@ let x1 = C.M(arg1 = 3, arg2 = 4, arg3 = 5)
 /// This is x2
 let x2 = C.M(arg1 = 3, arg2 = 4, ?arg3 = Some 5)
 
+/// This is x3
```
Comment (auduchinok) [inline tests/service/MultiProjectAnalysisTests.fs:47]: Could there also be a case with a multi-line comment?

File: tests/service/MultiProjectAnalysisTests.fs [hunk 2]
```diff
@@ -17,6 +17,7 @@ open System.IO
 
 open FSharp.Compiler.Service.Tests.Common
 
+let toIList (x: _ array) = x :> System.Collections.Generic.IList<_>
```
Comment (cartermp) [inline tests/service/MultiProjectAnalysisTests.fs:20]: nit: `open System.Collection.Generic` at the top of the file

**Comment by auduchinok** [comment]
A failure here:
https://github.com/dotnet/fsharp/blob/7780cab8dcaaa042c31b0588deed3d408af0d210/vsintegration/tests/UnitTests/LegacyLanguageService/Tests.LanguageService.QuickInfo.fs#L2979

@cartermp Is it OK that the test case seems to run (and fail) twice?

**Comment by mcon** [comment]
I'm getting a build error when trying to build the failing unit tests in question on master, is anyone able to suggest what the problem might be?

`C:\Users\Matt\fsharp\tests\FSharp.Build.UnitTests\FSharp.Build.UnitTests.fsproj : error MSB4057: The target "_GetRestoreSettingsPerFramework" does not exist in the project.`

**Comment by cartermp** [comment]
@auduchinok Probably not, no 🙂

**Comment by cartermp** [comment]
@mcon do you see that error when you build in VS?

**Comment by mcon** [comment]
When I use the command line `build.cmd` - I've got VS installed, but don't use use it as my IDE. Could be something to do with the fact I'd also built fcs in the same workspace - maybe new workspace time for me...

**Comment by mcon** [comment]
So, I've taken a look here, and the secondary problem (which I've now resolved) was that the "grab point" for bindings was after the `headBindingPattern`, and so the XML lines which were being picked up included XML comments on the function arguments (which, for a tooltip on a function, I don't think is desirable). I've resolved this by putting a grab point at the start of the `headBindingPattern`, so that only the XML comment above the binding is picked up, and then one at the end of the pattern, so that any XML comments on arguments aren't picked up erroneously by other bindings down the file.

I wanted to have the XML comments on the arguments show up in tooltips too, but though I was able to add the grab points in the right place, I wasn't able to work out where to store the comments in such a way that they could be retrieved later, after parsing, by the tooltip. As a result I've not included this enhancement, and have just stuck to fixing the bug.

Worth noting that prior to this change, XML comments on the function arguments didn't show up in the tooltips - this behaviour isn't altered as part of this PR.

**Comment by cartermp** [comment]
@mcon makes sense! Just fixing the bug here is definitely fine. Thanks!

**Pull request merged**